Ff104 HTMLElement.focus() - add focusVisible option (#19311)
mdn/content: files/en-us/web/api/htmlelement/focus/index.md

---
{{ APIRef("HTML DOM") }}

The **`HTMLElement.focus()`** method sets focus on the specified element, if it can be focused.
The focused element is the element that will receive keyboard and similar events by default.

By default the browser will scroll the element into view after focusing it, and it may also provide visible indication of the focused element (typically by displaying a "focus ring" around the element).

> Parameter options are provided disable the default scrolling and force visible indication on elements.

Reviewer: ```suggestion
Parameter options that are provided disable the default scrolling and force visible indication on elements.
```

or 

```suggestion
Parameter options are provided and can disable the default scrolling and force visible indication on elements.
```

Author: Good spot. Special option 3

```suggestion
Parameter options are provided to disable the default scrolling and force visible indication on elements.
```

- `options` {{optional_inline}}

  - : An optional object for controlling aspects of the focusing process.
    This object may contain the following properties:

    - `preventScroll` {{optional_inline}}
      - : A boolean value indicating whether or not the browser should scroll the document to bring the newly-focused element into view.
        A value of `false` for `preventScroll` (the default) means that the browser will scroll the element into view after focusing it.
        If `preventScroll` is set to `true`, no scrolling will occur.
    - `focusVisible` {{optional_inline}} {{experimental_inline}}

>       - : A boolean value that should be set to `true` to force visible indication that the element is focused.

Reviewer: What's the default value for `focusVisible`?

Author: That was answered in the following line "If the property is not `true` or not specified". I'll reword to say "by default"

>         If the property is not `true` or not specified, a browser may still provide visible indication if it determines that this would improve accessibility for users.

Author: ```suggestion
        By default, or if the property is not `true`, a browser may still provide visible indication if it determines that this would improve accessibility for users.
```
